mobile_packet_verifier/src/banning/ingester.rs

```diff
@@ -0,0 +1,63 @@
+use file_store::mobile_ban::{VerifiedBanReportSource, VerifiedBanReportStream};
+use futures::{FutureExt, StreamExt};
+use sqlx::PgPool;
+use task_manager::ManagedTask;
+
+pub struct BanIngestor {
+    pool: PgPool,
+    report_rx: VerifiedBanReportSource,
+}
+
+impl ManagedTask for BanIngestor {
+    fn start_task(
+        self: Box<Self>,
+        shutdown: triggered::Listener,
+    ) -> futures::future::LocalBoxFuture<'static, anyhow::Result<()>> {
+        self.run(shutdown).boxed_local()
+    }
+}
+
+impl BanIngestor {
+    pub fn new(pool: PgPool, report_rx: VerifiedBanReportSource) -> Self {
+        Self { pool, report_rx }
+    }
+
+    async fn run(mut self, mut shutdown: triggered::Listener) -> anyhow::Result<()> {
+        tracing::info!("starting ban ingestor");
+
+        loop {
+            tokio::select! {
+                _ = &mut shutdown => break,
```

Comment:
make the select biased, so shutdown is selected if triggered

Reply:
Good catch